test/e2e/apps/rc.go

```diff
@@ -172,6 +185,21 @@ var _ = SIGDescribe("ReplicationController", func() {
 		rcStatus, err := f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).Patch(context.TODO(), testRcName, types.StrategicMergePatchType, []byte(rcStatusPatchPayload), metav1.PatchOptions{}, "status")
 		framework.ExpectNoError(err, "Failed to patch ReplicationControllerStatus")
 		framework.ExpectEqual(rcStatus.Status.ReadyReplicas, int32(0), "ReplicationControllerStatus's readyReplicas does not equal 0")
+		ginkgo.By("waiting for RC to be modified")
+		for watchEvent := range rcWatchChan {
+			if watchEvent.Type == watch.Modified {
```

Comment:
same question about the intent of this loop... what is it intending to wait for?